### PR TITLE
cargo-release changes and migrate to windows-sys

### PR DESCRIPTION
**Pull request by sn99**
Changes include migrating to windows-sys and cargo release optimizations

### CODE REVIEW

**Comment by Narsil** [comment]
No.

(You didn't argument for this change, so I won't argument in my refusal)

**Comment by sn99** [comment]
@Narsil My bad, the changes include:

- Migrating from `winapi` to `windows-sys`: `winapi` is no longer maintained in favor of [`windows-rs`](https://github.com/microsoft/windows-rs) which is provided by Microsoft itself. It should provide more robustness against future windows upgrade and better compile and runtime guarantees
- `cargo.toml` release arguments especially `lto=true` help libs further down the line to have certain optimizations enabled

**Comment by Narsil** [comment]
I'll reopen, since it does seem `winapi` is not well maintained.

**Comment by sn99** [comment]
I am on a windows system and the last test seems to fail even on my machine, even before I made any changes:

```cmd
PS C:\Users\sn99\Downloads\rdev-main\rdev-main> cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.06s
     Running unittests src\lib.rs (target\debug\deps\rdev-c607d680501d202a.exe)

running 2 tests
test tests::test_keyboard_state ... ok
test windows::keycodes::test::test_reversible ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests\listen_and_simulate.rs (target\debug\deps\listen_and_simulate-7e3563f7acb512d4.exe)

running 1 test
test test_listen_and_simulate ... FAILED

failures:

---- test_listen_and_simulate stdout ----
thread 'test_listen_and_simulate' panicked at 'assertion failed: `(left == right)`
  left: `MouseMove { x: 0.0, y: 1.0 }`,
 right: `MouseMove { x: 0.0, y: 0.0 }`', tests\listen_and_simulate.rs:39:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    test_listen_and_simulate

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.10s

error: test failed, to rerun pass `--test listen_and_simulate`
```

**Review by Narsil — COMMENTED** [review]
Ok gave it a  little more thorough review.

I'm being harsh. This is not towards you, mostly towards Microsoft.
Since I had to go through all the BS of their API, I'm very unfavorable of touching anything.

Despite the outdatedness of `winapi`  I actually went through **a ton** of testing to make sure things work. I'm very reluctant to make any changes that change types.

And there's a least 2 instances here where type seem to be modified between winapi and windows-rs. This in addition with the lack of consistency with the original windows docs, give me very little confidence that those modifications are actually working.

Of course I'm likely to be wrong, since `windows-rs`  seems to be getting quite a lot of usage, but my experience, is that this crate touches not so widely used API, or in an odd fashion (or many devs are getting insane).

All in all, I'd like to keep this PR open. **It's a very nice PR, thanks for working on it.**
But I'm not trusting enough that the changes are actually valid to merge it anytime soon. 

This will happen only after extensive testing (especially the corner cases which I know caused a ton of issues). And that unfortunately won't happen anytime soon.
This is now a personal project, and it's already in a working state, so I'd rather not spend the time on something I don't think adds a ton of values (like fixing existing bugs, or add new features)

File: examples/channel.rs [hunk 1]
```diff
@@ -9,14 +9,14 @@ fn main() {
         listen(move |event| {
             schan
                 .send(event)
-                .unwrap_or_else(|e| println!("Could not send event {:?}", e));
+                .unwrap_or_else(|e| println!("Could not send event {e:?}"));
```
[the remark this answers is not in the document]
Comment (Narsil) [reply examples/channel.rs:12]: I see it's a new clippy lint. Let's still move it to another PR.

**Comment by sn99** [comment]
I think the suggested changes are pretty good, I will look into them again tomorrow, `windows-rs` has 2 parts `windows` and `windows-sys`, the differences are mentioned here https://kennykerr.ca/rust-getting-started/windows-or-windows-sys.html, do you want me to choose `windows` rather than `windows-sys`?

**Comment by Narsil** [comment]
> I think the suggested changes are pretty good, I will look into them again tomorrow, `windows-rs` has 2 parts `windows` and `windows-sys`, the differences are mentioned here https://kennykerr.ca/rust-getting-started/windows-or-windows-sys.html, do you want me to choose `windows` rather than `windows-sys`?


I think as a starter let's keep to `windows-sys`. It's closer to what is already there so it will be easier to spot the core differences.

**Comment by sn99** [comment]
I will make separate pulls

**Pull request closed without merging**